pkg/pub_integration/lib/src/headless_env.dart

```diff
@@ -12,15 +12,15 @@ import 'package:puppeteer/puppeteer.dart';
 /// Creates and tracks the headless Chrome environment, its temp directories and
 /// and uncaught exceptions.
 class HeadlessEnv {
-  final String testName;
+  final String? _testName;
   final String _origin;
   final String? _coverageDir;
   final Directory _tempDir;
-  final bool debug;
+  final bool _displayBrowser;
   Browser? _browser;
-  final clientErrors = <ClientError>[];
-  final serverErrors = <String>[];
-  late final bool trackCoverage =
+  final _clientErrors = <ClientError>[];
+  final _serverErrors = <String>[];
```

Comment:
Unrelated to this PR, but shouldn't these be scoped to `withPage` ?

What if there is concurrent invocations of `withPage`, or what if an invocation fails and the test case ignores the error and makes another `withPage` invocation (then that'll fail for no reason).

Reply:
Good point, moving them into the withPage context.